pcsx2-qt/EmuThread.cpp

```diff
@@ -889,4 +890,20 @@ DEFINE_HOTKEY("ToggleFullscreen", "General", "Toggle Fullscreen", [](bool presse
 	if (!pressed)
 		g_emu_thread->toggleFullscreen();
 })
+// Input Recording Hot Keys
+DEFINE_HOTKEY("InputRecToggleMode", "Input Recording", "Toggle Recording Mode", [](bool pressed) {
+	if (!pressed) // ?? - not pressed so it is on key up?
+	{
+		g_InputRecordingControls.RecordModeToggle();
+	}
+})
+// TODO - Vaser - the way the pause/resuming used to work is broken on Qt, needs a rewrite. 
+//	- Currently if you frame advance you can't get out of frame advancing!
+DEFINE_HOTKEY("InputRecFrameAdvance", "Input Recording", "Frame Advance", [](bool pressed) {
+	if (!pressed) // ?? - not pressed so it is on key up?
+	{
+		g_InputRecordingControls.FrameAdvance();
```

Comment:
IMO, frame advance shouldn't be part of input recording, it has use cases outside of it as well.

Reply:
removed as you are apparently PR'ing the feature.  I agree, and im glad a proper pause feature has also been added to the main emulator's functions.